### PR TITLE
Add fn to automatically compute padding

### PR DESCRIPTION
**Pull request by ashnair1**
#### Changes

Previously, users were required to compute the padding themselves to ensure chaining of `ExtractTensorPatches` and `CombineTensorPatches` produced the expected result. This PR adds a function to automatically compute the required padding.

Before:

```
import torch
from kornia.contrib import CombineTensorPatches, ExtractTensorPatches

# Original size
orig = (12, 23)
# Window size
win = (7, 7)
padding = (1, 1, 5, 0) # or (1, 1, 0, 5) / (1, 1, 2, 3) / (1, 1, 3, 2) / (1, 1, 4, 1) / (1, 1, 1, 4)
image = torch.randn(1, 1, orig[0], orig[1])
tiler = ExtractTensorPatches(window_size=win, stride=win, padding=padding)
merger = CombineTensorPatches(original_size=orig, window_size=win, unpadding=padding)
image_tiles = tiler(image)
new_image = merger(image_tiles)
print(new_image.shape)
assert (image == new_image).all()
```

With `compute_padding`:

```
import torch
from kornia.contrib import CombineTensorPatches, ExtractTensorPatches, compute_padding

# Original size
orig = (12, 23)
# Window size
win = (7, 7)
padding = compute_padding(orig_size, win)
image = torch.randn(1, 1, orig[0], orig[1])
tiler = ExtractTensorPatches(window_size=win, stride=win, padding=padding)
merger = CombineTensorPatches(original_size=orig, window_size=win, unpadding=padding)
image_tiles = tiler(image)
new_image = merger(image_tiles)
print(new_image.shape)
assert (image == new_image).all()
```

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [x] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: kornia/contrib/extract_patches.py [hunk 1]
```diff
@@ -8,6 +8,55 @@
 PadType = Union[Tuple[int, int], Tuple[int, int, int, int]]
 
 
+def compute_padding(
+    original_size: Union[int, Tuple[int, int]], window_size: Union[int, Tuple[int, int]]
+) -> Tuple[int, int, int, int]:
+    r"""Compute required padding to ensure chaining of :class:`~kornia.contrib.ExtractTensorPatches` and
+    :class:`~kornia.contrib.CombineTensorPatches` produces expected result.
```
Comment (ashnair1) [inline kornia/contrib/extract_patches.py:15]: Wasn't sure if I should refer to the class `ExtractTensorPatches` or the function `extract_tensor_patches`

File: kornia/contrib/extract_patches.py [hunk 2]
```diff
@@ -8,6 +8,55 @@
...
+        >>> out = extract_tensor_patches(image, window_size=(3, 3), stride=(3, 3), padding=padding)
+        >>> combine_tensor_patches(out, original_size=(4, 3), window_size=(3, 3), stride=(3, 3), unpadding=padding)
+        tensor([[[[ 0,  1,  2],
+                  [ 3,  4,  5],
+                  [ 6,  7,  8],
+                  [ 9, 10, 11]]]])
+
+    .. note::
+        This function is supposed to be used in conjunction with :func:`extract_tensor_patches`
+        and :func:`combine_tensor_patches`.
```
Comment (ashnair1) [inline kornia/contrib/extract_patches.py:36]: Same as above.

**Review by ducha-aiki — COMMENTED** [review]
Overall looks good to me.
One thing - it makes sense to combine this with compute_padding function used in kornia.filers https://github.com/kornia/kornia/blob/master/kornia/filters/filter.py#L10

File: kornia/contrib/extract_patches.py (hunk 1 above)
Comment (edgarriba) [reply kornia/contrib/extract_patches.py:15]: yeah, I'd prioritize functional api

**Review by edgarriba — COMMENTED** [review]
possible it makes sense to add this under `kornia.utils` and combine/use the function that @ducha-aiki pointed out.

**Comment by ashnair1** [comment]
I took a look at the aforementioned function and don't think there is an overlap in functionality.

- The function in this PR aims to calculate minimum padding required for extraction and combining to work correctly. Calculation is based on two params original_image_size and window_size used in those functions. The function in `filters.py` is used for padding an ND kernel.
- padding format is different. The function in this PR produces `(padtop, padbottom, padleft, padright)` while `_compute_padding` produces `(padleft, padright, padtop, padbottom)`.  While this is just a format issue it reflects their respective usecases. The former is meant to be passed to the extract and combine functions since its intuitive that the first value of original_size, window_size and padding refer to height while the second refers to width. But the latter is meant to be passed to `torch.nn.functional.pad` directly.

**Comment by edgarriba** [comment]
> @ducha-aiki @edgarriba
> 
> I took a look at the aforementioned function and don't think there is an overlap in functionality.
> 
> * The function in this PR aims to calculate minimum padding required for extraction and combining to work correctly. Calculation is based on two params original_image_size and window_size used in those functions. The function in `filters.py` is used for padding an ND kernel to constrain the shape of the output after it is applied to an input.
> * padding format is different. The function in this PR produces `(padtop, padbottom, padleft, padright)` while `_compute_padding` produces `(padleft, padright, padtop, padbottom)`.  While this is just a format issue it reflects their respective use cases. The former is meant to be passed to the extract and combine functions since its intuitive that the first value of original_size, window_size, stride and padding refer to height while the second refers to width and the functions resolve this before passing to `torch.nn.functional.pad` (Refer [here](https://github.com/kornia/kornia/blob/f967e7529dd85db263484ae7ccc027784c4ecaf0/kornia/contrib/extract_patches.py#L207) and [here](https://github.com/kornia/kornia/blob/f967e7529dd85db263484ae7ccc027784c4ecaf0/kornia/contrib/extract_patches.py#L286)). But the latter is meant to be passed to `torch.nn.functional.pad` directly since it follows the torch padding convention.
> 
> I can rename the function if that reduces confusion.

I believe your proposal it’s fine for now. Later we can adapt the one in filter to use the one in this PR ? In that case I’d just make the parameters names more "generic ".  But overall LGTM
